DDS-862_app_status_errors

dmlond opened the pull request:
should work correctly now

dmlond committed:
changed to an example of an OpenidAuthenticationService

dmlond committed:
moved pry.byebug into its own section for :docker, :test
this allows us to interact with pry within a running server instance. Instead of
docker-compose up -d server
run
docker-compose run --rm --service-ports server

dmlond committed:
added rabbitmq to line that launches dependant services

dmlond committed:
removed spurious status[:graphdb] entry
Uses ApplicationJob.conn instead (which starts the connection and returns it)

dmann commented on Gemfile:
```diff
@@ -72,14 +72,17 @@ group :development, :docker, :test do
   gem 'rspec-rails'
 end
 
+group :docker, :test do
```
Would it make sense to add `:development` to this group? I'm not sure how we'd use it from the rails console, since we wouldn't want to commit code with `binding.pry` in it... Hmm, maybe just ignore this thought then.  :P

dmann commented on app/api/dds/v1/app_api.rb:
```diff
@@ -60,7 +59,7 @@ class AppAPI < Grape::API
               ApplicationJob.distributor_exchange_name
             ].each do |expected_exchange|
-              unless Sneakers::CONFIG[:connection].exchange_exists? expected_exchange
+              unless ApplicationJob.conn.exchange_exists? expected_exchange
```
:wrist_slap: ApplicationJob.conn is a private method...

Also, if the problem is that the connection isn't started, there should be a test that demonstrates that.  :P

dmann commented on app/api/dds/v1/app_api.rb:
```diff
@@ -76,7 +75,7 @@ class AppAPI < Grape::API
             ].concat(application_job_workers)
             .each do |expected_queue|
-              unless Sneakers::CONFIG[:connection].queue_exists? expected_queue
+              unless ApplicationJob.conn.queue_exists? expected_queue
```
BTW, if you want ApplicationJob.conn to be public, you'll have to write some unit tests for it.

dmann reviewed (changes requested):
It would be good to have some tests to justify these changes...

dmlond committed:
created test to ensure that start is called on the connection

job_shared_examples
  mocks Sneakers::CONFIG[:connection] instead of BunnyMock::Session
  so that we can test with TEST_WITH_BUNNY=1

app_api starts the connection

dmlond commented:
wrote some tests

dmlond committed:
Merge branch 'develop' into DDS-862_app_status_errors
